app/assets/javascripts/galleries/uploader.js

```diff
@@ -40,6 +42,17 @@ function bindFileInput(fileInput, form, submitButton, formData) {
     imageMaxHeight: 400,
 
     add: function(e, data) {
+      if (typeof limit !== 'undefined' && limit > 1) {
+        var isFirstFile = (fileInput[0].files[0] == data.files[0]);
+        var numUploading = fileInput[0].files.length;
+        if (isFirstFile) numFiles += numUploading;
```

Comment:
Don't want to let them upload limit *at a time*, we want to let them upload limit *total*.

Reply:
Does this safely handle users uploading 5 and then 150 files? Looks like it'll add 5 files, so `fileInput[0].files` has a bunch in, and then when it runs with the next 150, isFirstFile will be false (`fileInput[0].files` is presumably just concatenated with the new 50, and so `fileInput[0].files[0] != newFile`, so we don't add more here, so we never trigger the `numFiles > limit` thingy)?

I'm thinking this might be simpler if we just add to `numFiles` each time we iterate this, and then we cut off once it gets to 150. Possibly less straightforward for the user, because we may not upload in order, but we still limit to 150 on the page. I think we might already be providing similar behavior, but I'm not sure.

Reply:
It does in fact safely handle uploading 5 and then 150, in that the 5 will still have succeeded and then it will error on the second round. I tested a bunch with limit set to 3 and upload 2 and then 2 again (failed) and then 1 (worked), etc etc.